superset/utils/decorators.py

```diff
@@ -89,13 +94,26 @@ def wrapper(*args: Any, **kwargs: Any) -> ETagResponseMixin:
                         raise
                     logger.exception("Exception possibly due to cache backend.")
 
+            # if cache is stale?
+            if get_latest_changed_on:
+                content_changed_time = get_latest_changed_on(*args, **kwargs)
+                if (
+                    response
+                    and response.last_modified
+                    and response.last_modified.timestamp()
+                    < content_changed_time.timestamp()
+                ):
+                    response = None
+            else:
+                content_changed_time = datetime.utcnow()
```

Comment:
We use this `content_changed_time` as cache's last_modified time.
for dashboard `content_changed_time` is dashboard entity's latest updated time (like metadata, chart metadata changed time etc). this data is from a callback function.
for explore_json, the cache is query results and there is no entity's latest modified time to use. so we use request time (now) as cache's last_modified time.

Reply:
I know generalizing too soon is not a good practice, but I wonder if we should pass a callable called `is_stale` here. It would simplify the decorator logic, and since it would be defined closer to the dashboard it might simplify the logic there as well.

Reply:
at first i thought is_stale is a good idea. but when start refactor it, i found last_modified time is needed by decorator function(to set header). So is_stale (only boolean value) is not enough. 
So instead of using is_stale return true or false, I prefer to keep get_last_modified, and use last_modified time to invalid cache.